core/static/js/dit.class.modal.js

```diff
@@ -237,10 +246,15 @@
     self.$content.append(content);
   }
 
-  // Tries to add focus to the first found element allowed nwith atural focus ability.
+  // Tries to add focus to the first found element allowed with natural focus ability.
   Modal.prototype.focus = function() {
     var self = this;
-    self.$content.find("video, a, button, input, select").eq(0).focus();
+    // If $firstFocusElement didn't find anything
```

Comment:
Understand that you're being defensive with this check, and also the comment about appending elements, but would argue it is not handled in the best way. 

You're right about re-finding the first focus element after appending, but we should being doing something like:

1. Run `this.$firstFocusElement = Modal.findFirstFocusElement($container);` as another action of `Modal.prototype.setContent()` function
2. Add some kind of `this.$firstFocusElement.off()` (and for lastFocusElement) as opening (reset) lines for `Modal.bindKeyboardFocusEvents()` function 
3. Call `Modal.bindKeyboardFocusEvents()` as a last action to the `Modal.prototype.setContent()` function

That would clear things up and leave this function as simple and `self.$firstFocusElement.focus()`  
By all means leave the `!self.$firstFocusElement.length` check in if you want but I think, if `Modal.findFirstFocusElement` returns an empty jQuery object (so length ==0), then jQuery also handles a `.focus()` chained call without error as well, meaning the check isn't actually required. 

You can test that last assumption to be sure but, if you run this...
`$().focus()`
...in your browser console, you should see no JS error. That should be the same thing.